src/systems/tile_update.rs

```diff
@@ -0,0 +1,143 @@
+use amethyst::assets::{AssetStorage, Loader};
+use amethyst::core::cgmath::{Deg, Vector3};
+use amethyst::core::specs::prelude::{Read, ReadExpect, ReadStorage, System, Write, WriteStorage};
+use amethyst::core::transform::Transform;
+use amethyst::renderer::{
+    Material, MaterialDefaults, Mesh, MeshHandle, ObjFormat, PngFormat, Texture, TextureMetadata,
+};
+use assetloading::asset_loader::AssetManager;
+use entities::tile::{LevelGrid, Tile};
+use std::vec::IntoIter;
+
+pub struct TileUpdateSystem;
+
+impl<'a> System<'a> for TileUpdateSystem {
+    type SystemData = (
+        Write<'a, TileUpdateQueue>,
+        Read<'a, Vec<([[Tile; 3]; 3], String)>>,
+        Read<'a, LevelGrid>,
+        ReadStorage<'a, Tile>,
+        ReadExpect<'a, Loader>,
+        WriteStorage<'a, Transform>,
+        (
+            Write<'a, AssetManager<Mesh>>,
+            WriteStorage<'a, MeshHandle>,
+            Write<'a, AssetStorage<Mesh>>,
+        ),
+        (
+            Write<'a, AssetManager<Texture>>,
+            WriteStorage<'a, Material>,
+            Write<'a, AssetStorage<Texture>>,
+            ReadExpect<'a, MaterialDefaults>,
+        ),
+    );
+
+    fn run(
+        &mut self,
+        (
+            mut tiles,
+            dict,
+            level_grid,
+            tile_storage,
+            amethyst_loader,
+            mut transform_storage,
+            (mut mesh_manager, mut mesh_handles, mut mesh_storage),
+            (mut tex_manager, mut mat_storage, mut tex_storage, mat_defaults),
+        ): Self::SystemData,
+    ) {
+        if tiles.is_empty() {
+            return;
+        };
+
+        // TODO setup stuff
+        // TODO create a grid here and pass it... unnescessary to create it multiple times in one tick
+
+        for (x, y) in tiles.clone().into_iter() {
+            // get sprite defintion
+            let (wall_type, wall_rotation) =
+                level_grid.determine_sprite_for(x, y, &dict, &tile_storage);
+
+            let mut transform = Transform::default();
+            // maybe set 00 top left instead of bottom left?
```

Comment:
please either do that and remove the comment or don't do it and remove the comment :)

Reply:
you tell me